docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

```diff
@@ -334,6 +334,8 @@ If you are extending the server's REST APIs through the following SPIs:
 You need to add an empty `META-INF/beans.xml` to the JAR file where your custom providers are packaged. Otherwise, they are not recognized by the server
 at runtime.
 
+If you are using `RealmResourceSPI` or `AdminRealmResourceSpi`, another alternative to adding an empty `beans.xml` under `META-INF` is annotating the JAX-RS resource classes with the annotation `jakarta.ws.rs.ext.Provider`.
```

Comment:
```suggestion
If you are using `RealmResourceSPI` or `AdminRealmResourceSpi`, another alternative to add an empty `beans.xml` under `META-INF` is annotating the JAX-RS resource classes with the annotation `jakarta.ws.rs.ext.Provider`.
```

Reply:
Hi @andymunro, I am not 100% sure but I think here actually the verb form ending with `-ing` is correct because here `to` is used as preposition. The meaning that I want to convey is:
Instead of adding an empty `beans.xml`, there is also the possibility of annotating the JAX-RS resource classes with  the annotation `jakarta.ws.rs.ext.Provider`, if you are using `RealmResourceSPI` or `AdminRealmResourceSpi`.

Reply:
I found something about it https://blog.harwardcommunications.com/2012/03/30/when-is-it-correct-to-use-to-plus-ing-part-1/